### ISSUE TITLE
Add method to get longitude/latitude for Accounts

### ISSUE THREAD

**Issue by pbrondum** [mike4aday/SwiftlySalesforce#39]

**Comment by mike4aday** [comment]
Delivered in version 4.0.0. See `Address` struct.

**Comment by pbrondum** [comment]
Spelling mistake in Address.

case block = "Blcok"

**Comment by mike4aday** [comment]
@pbrondum -- thanks. I'll fix that shortly; opened issue #44

**Comment by pbrondum** [comment]
Mike - Having problems with the Address feature. I'm trying to get
lati/long coordinates.

Tried:

  }.then {

            let AccSoql = "SELECT ShippingAddress FROM Account"

            return salesforce.query(soql: AccSoql)

        }.then {

            (result: QueryResult) -> () in

            print(result)

            let addresses = result.records.map { Address(json: $0 as [String
: AnyObject]) }


I have activated the geo clean rule for accounts.  Still all the fields
(including long/lat, except for street)i are null.


*...}], ["ShippingAddress": {*

*    city = "<null>";*

*    country = "<null>";*

*    countryCode = "<null>";*

*    geocodeAccuracy = "<null>";*

*    latitude = "<null>";*

*    longitude = "<null>";*

*    postalCode = "<null>";*

*    state = "<null>";*

*    stateCode = "<null>";*


*    street = "345 Shoreline Park\nMountain View, CA ....*


Per

Regards,

Per Jakobsen

Regular mail : perbrondum@gmail.com
Secure Mail: isper@protonmail.com

**Comment by mike4aday** [comment]
@pbrondum do you see the latitude and longitude values in the Salesforce UI or via Workbench - the clean rules can take time? Also It looks like your map function is getting the record "root," rather than the Address JSON. Argument should be $0.address(for: "ShippingAddress")

**Comment by mike4aday** [comment]
@pbrondum I added an example [here](https://github.com/mike4aday/SwiftlySalesforce#example-retrieve-an-accounts-billing-address)

**Comment by pbrondum** [comment]
No matter how I use your example I get:
Value of type 'Address?' has no member 'longitude'

**Comment by mike4aday** [comment]
@pbrondum can you post your code?

**Comment by pbrondum** [comment]
import UIKit
import MapKit
import SwiftlySalesforce
import PromiseKit

final class MasterViewController: UITableViewController {
    var locManager = CLLocationManager()
    var currentLocation: CLLocation!

	override func viewDidLoad() {
		super.viewDidLoad()
		refreshControl?.addTarget(self, action: #selector(MasterViewController.handleRefresh), for: UIControlEvents.valueChanged)
        locManager.requestWhenInUseAuthorization()
        
        if (CLLocationManager.authorizationStatus() == CLAuthorizationStatus.authorizedWhenInUse ||
            CLLocationManager.authorizationStatus() == CLAuthorizationStatus.authorizedAlways){
            currentLocation = locManager.location
            print("Curr Lat: \(currentLocation.coordinate.latitude)")
            print("Curr Long: \(currentLocation.coordinate.longitude)")
            print(currentLocation)
        }
        
        first {
            salesforce.retrieve(type: "Account", id: "00130000007dp2SAAQ")
        }.then {
            record in
            let address = record.address(for: "BillingAddress")
            let longitude = address.longitude
            let latitude = address.latitude
            let AccountLocation = CLLocation(latitude: latitude, longitude: longitude)
            let distanceFromCurrentLocation = self.currentLocation.distance(from: AccountLocation) // in meters
            print("Distance in Meters : \(distanceFromCurrentLocation)")
        }.always {
            print("Got here")
        }
        loadData(refresh: true)
	}
	
	override func viewWillAppear(_ animated: Bool) {
		super.viewWillAppear(animated)
		//loadData(refresh: false)
	}
	
	override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
		if let destinationVC = segue.destination as? DetailViewController,
			let cell = sender as? UITableViewCell,
			let indexPath = tableView.indexPath(for: cell),
			let task = TaskStore.shared.cache?[indexPath.row] {
			destinationVC.task = task
			destinationVC.title = task.subject ?? ""
		}
	}

**Comment by mike4aday** [comment]
@pbrondum sorry, `record.address(for: "BillingAddress")` returns an optional address, i.e. `Address?`.

I updated the example code in the README. It now reads:
```swift
record in
if let address = record.address(for: "BillingAddress") {
    let longitude = address.longitude
    let latitude = address.latitude
    // You could put a marker on a map...
}
```

Does that work?

**Comment by pbrondum** [comment]
Now I get the following for both assignments:
'longitude' is inaccessible due to 'internal' protection level

**Comment by mike4aday** [comment]
@pbrondum sorry again... the member vars should be `public` -- I'll fix ASAP.

**Comment by pbrondum** [comment]
Easy fix. No rush. Thanks. 

Per

**Comment by mike4aday** [comment]
@pbrondum should be fixed now in version 4.0.3 - `Address` members are explicitly `public`. 
Thanks
